os/kernel/task/task_setup.c

```diff
@@ -497,28 +497,36 @@ static int thread_schedsetup(FAR struct tcb_s *tcb, int priority, start_t start,
 		up_initial_state(tcb);
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
-		/* Copy the parent task ram details to this task */
+		/* Copy some attributes from parent task to this task */
 		rtcb = this_task();
-		tcb->uspace = rtcb->uspace;
-		tcb->uheap = rtcb->uheap;
 		tcb->app_id = rtcb->app_id;
+
+		/* If a kernel thread is getting created due to some
+		 * call from a user task or pthread, then the kernel thread
+		 * will not inherit the some attributes related to heap, uspace,
+		 * memory protection, etc..
+		 */
+		if ((tcb->flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL) {
+			tcb->uspace = rtcb->uspace;
```

Comment:
These members will be set when it's the user thread. Could you verify all members which are set in the if statement are used after null or 0 checking?
I mean, 
`uspace` is used when it's not null in all codes?
same in uheap
same in pgtbl
and so on

Reply:
Thanks Chang. I verified and added null checks where  they are required.